### PR TITLE
mongo, nosql, glabel silent installation

### PR DESCRIPTION
**Pull request by sachinnagpal**

### CODE REVIEW

File: src/backend/utils.js [hunk 1]
```diff
@@ -94,12 +95,22 @@ function sleep(ms) {
   return new Promise(resolve => setTimeout(resolve, ms));
 }
 
+async function extractFiles(sourceFolderPath, destinationFolderPath) {
+  try {
+    await decompress(sourceFolderPath, destinationFolderPath)
```
Comment (riturajborpujari) [inline src/backend/utils.js:100]: Can we use a PS1 command `Extract-Archive` instead?
Additional libs bloat up the size

File: src/backend/installer.js [hunk 2]
```diff
@@ -258,25 +322,40 @@ const appRuning = function (dependencyFileName) {
...
-    utils.runSpawnCommand(command)
+  try {
+    let env = "prod"
+    let logLevel = `info`
+    let appJson = fs.readJSONSync(path.join(constants.DIRECTORIES.APP, "app.json"))
+    let nssmServices = appJson.filter(app => app.type === constants.DEPENDENCY_TYPES.NSSM_SERVICE)
+    log.info(JSON.stringify(nssmServices), null, 2)
+    utils.sendAlert(constants.UI_ALERT_KEYS.ENVIRONMENT_SETUP, { configuring: true })
+    for (let service of nssmServices) {
+      let command = `nssm set ${service.name} AppEnvironmentExtra NODE_ENV=${env} O4S_CONFIG_PATH=${path.join(constants.DIRECTORIES.APP, "config.json")} LOG_LEVEL=${logLevel} >$null 2>&1`
```
Comment (riturajborpujari) [inline src/backend/installer.js:333]: Where are the output routing commands for nssm services???

File: src/backend/installer.js [hunk 3]
```diff
@@ -150,14 +151,67 @@ const registerService = function (dependency) {
...
+  log.info(JSON.stringify(installCommandRes))
+  if (!installCommandRes.success) {
+    throw new Error(`Mongo Installation failed`)
+  }
+  log.info(`Mongo installed successfully`)
 }
 
+const createMongoPaths = async function () {
+  try {
+    const mongoDbPath = "C:\\Program Files\\MongoDB\\Server\\4.2"
```
Comment (riturajborpujari) [inline src/backend/installer.js:196]: Isn't this already available as a constant?

File: src/backend/installer.js [hunk 4]
```diff
@@ -150,14 +151,67 @@ const registerService = function (dependency) {
...
+  if (!mongoInstalled.success) {
+    return
   }
+  const uninstallCommand = `& "${constants.DIRECTORIES.MONGO}\\bin\\mongod.exe" --remove`
+  utils.runSpawnCommand(uninstallCommand)
+  const binDirectory = path.join(constants.DIRECTORIES.MONGO, "bin");
+  fs.rmSync(binDirectory, { recursive: true, force: true });
+  log.info(`Uninstalled mongo successfully`)
+}
+const installMongSyncWithoutPopup = async function () {
```
Comment (riturajborpujari) [inline src/backend/installer.js:167]: Why the `Sync` in the name when the function is clearly an asynchronous one?

File: src/backend/installer.js [hunk 5]
```diff
@@ -150,14 +151,67 @@ const registerService = function (dependency) {
...
+  uninstallMongo()
+  const mongoDbPath = constants.DIRECTORIES.MONGO
+  const unzippedFolderContent = path.join(mongoDbPath, "mongodb-win32-x86_64-2012plus-4.2.8")
```
Comment (riturajborpujari) [inline src/backend/installer.js:170]: This variable will hold the **path** for the unzipped content right?

File: src/backend/installer.js [hunk 6]
```diff
@@ -150,14 +151,67 @@ const registerService = function (dependency) {
...
+  const mongoDbConfigPath = path.join(mongoDbPath, "mongod.cfg")
+  await createMongoPaths()
```
Comment (riturajborpujari) [inline src/backend/installer.js:172]: this should be at the top of the method

File: src/backend/installer.js [hunk 7]
```diff
@@ -150,14 +151,67 @@ const registerService = function (dependency) {
...
+  const mongoZipPath = path.join(constants.DIRECTORIES.APP, "mongodb-win32-x86_64-2012plus-4.2.8.zip")
+  await utils.extractFiles(mongoZipPath, mongoDbPath)
+  log.info(`files are exctracted successfully `)
+  try {
+    fs.copySync(unzippedFolderContent, mongoDbPath, { overwrite: true })
```
Comment (riturajborpujari) [inline src/backend/installer.js:177]: What is this copying??
`unzippedFolderContent` already has the value `path.join(mongoDbPath, "mongodb-win32-x86_64-2012plus-4.2.8")`

What? Where from? Where to?

File: src/backend/installer.js [hunk 8]
```diff
@@ -150,14 +151,67 @@ const registerService = function (dependency) {
...
+    log.info("Files are copied")
+  } catch (err) {
+    log.info(`Failed to copy content`)
+    log.info(err)
+    throw err
+  }
+  fs.rmSync(unzippedFolderContent, { recursive: true, force: true });
+  let installCommand = `& "${mongoDbPath}\\bin\\mongod.exe" --config "${mongoDbConfigPath}" --install`
+  let installCommandRes = utils.runSpawnCommand(installCommand)
```
Comment (riturajborpujari) [inline src/backend/installer.js:186]: maybe extract this part to its own method `const installResult = runMongoInstallationCommand()`

File: src/backend/installer.js (hunk 7 above)
Comment (sachinnagpal) [reply src/backend/installer.js:177]: So we are moving the contents from directory `mongodbpath + mongodb-win32-x86_64-2012plus-4.2.8` to `mongoDbPath` 
bin directory basically

File: src/backend/installer.js (hunk 5 above)
Comment (sachinnagpal) [reply src/backend/installer.js:170]: Yes

File: src/backend/utils.js (hunk 1 above)
Comment (sachinnagpal) [reply src/backend/utils.js]: Done

**Review by riturajborpujari — APPROVED** [review]

**Pull request merged**